qiskit/visualization/timeline/layouts.py

```diff
@@ -0,0 +1,143 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+A collection of functions that decide the layout of an output image.
+
+There are 2 types of layout functions in this module.
+
+1. layout.bit_arrange
+
+In this stylesheet entry the input data is a list of `types.Bits` and returns a
+sorted list of `types.Bits`.
+
+
+The format of generator is restricted to:
+
+    ```python
+
+    def my_layout(bits: List[types.Bits]) -> List[types.Bits]:
+
+        # sort a list of bits
+        pass
```

Comment:
what is this here for? it looks contradictory to the function having a return value

Reply:
some code for sorting depending what user want to do. is there any good example to write abstract functionality?

Reply:
I think it would just be the comment with no implementation, or a `# your code here` or an ellipsis `# ...`